draft-kowalik-rpp-architecture-01.adoc

```diff
@@ -243,14 +243,16 @@ Other transform operations like renew, or restore which are not addressable reso
 This basic set of rules and guidelines will be further refined in the RPP specifications and give an universal toolset for extending RPP with new resources and commands.
 
 
-==== RPP specific error codes and relation to HTTP error codes
+==== RPP specific status codes and relation to HTTP status codes
 
-RPP utilizes both HTTP status codes and RPP-specific error codes within RPP-specific HTTP Headers and the response body for detailed error reporting and allowing the client or an intermediate to determine what action to take based on status code and header details only.
+RPP utilizes both HTTP status codes and RPP-specific status codes, using RPP-specific HTTP Headers. The response body may contain machine readable problem details <<RFC7807>> for detailed status reporting. Using HTTP status codes and headers allows the client or an intermediate (HTTP status only) to determine what action to take based on status code and header details only.
 
-*  Use of HTTP status codes to indicate general categories of errors (e.g., 2xx success responses, 4xx for client errors, 5xx for server errors) <<RFC7231>>.
+*  Use of HTTP status codes to indicate general status categories (e.g., 2xx success responses, 4xx for client errors, 5xx for server errors) <<RFC7231>>.
 *  Use of additional signalling already standardised for HTTP, for example for rate limiting
-*  Definition of RPP-specific error codes, warnings of additional processing information, provided in the response, preferably outside of resource representation (e.g. in HTTP Headers) to give granular information about provisioning errors.
-*  Categorization of RPP error codes as temporary or permanent to guide client retry behavior.
+*  Definition of RPP-specific status codes, warnings of additional processing information, provided in the response, preferably outside of resource representation (e.g. in HTTP Headers) to give granular information about provisioning request status. 
+* RPP-specific status codes will be defined in a way that allows for easy mapping to HTTP status codes, but also provide additional information specific to RPP operations.
+* Both HTTP and RPP-specific status codes are used to determine if the outcome of a request is successful or not.

```

Comment:
If you say "both" does it mean really that the information is scattered between the two?
I rather though of a setup, where:
- HTTP code would be sufficient to know whether the operation was successful or not
- RPP code would deliver more detailed information, but generally would also carry the success/fail information
- there might be a situation of (potentially error) response occuring purely on HTTP level, without any RPP code. We likely would need to be specific of situations where this would be the case (example: authentication or authorization errors, which may be covered on a level outside of RPP)